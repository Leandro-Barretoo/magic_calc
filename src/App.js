import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import Calculator from './components/Calculator';
import Navigation from './components/Navigation';
import Home from './components/Home';
import Quote from './components/Quote';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Switch>
        <Route exact path="/math__magicians">
          <Navigation />
          <Home />
        </Route>
        <Route path="/math__magicians/calculator">
          <Navigation />
          <Calculator />
        </Route>
        <Route path="/math__magicians/quote">
          <Navigation />
          <Quote />
        </Route>
        <Redirect to="/math__magicians" />
      </Switch>
    );
  }
}

export default App;
