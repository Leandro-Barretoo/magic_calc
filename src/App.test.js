import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import App from './App';

describe('Testing the App component', () => {
  it('should render correctly to the dom', () => {
    const quote = renderer.create(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    ).toJSON();
    expect(quote).toMatchSnapshot();
  });
});
