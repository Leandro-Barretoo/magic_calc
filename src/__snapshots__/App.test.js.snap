// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App component should render correctly to the dom 1`] = `
Array [
  <div
    className="nav-container"
  >
    <span
      className="logo"
      data-testid="logo"
    >
      <a
        className="Visited"
        href="/math__magicians"
        onClick={[Function]}
      >
        Math Magicians
      </a>
    </span>
    <ul
      className="link-container"
    >
      <li
        className="links"
      >
        <a
          className="Visited"
          data-testid="home"
          href="/math__magicians"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="links"
      >
        <a
          className="Visited"
          data-testid="calculator"
          href="/math__magicians/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="links"
      >
        <a
          className="Visited"
          data-testid="quote"
          href="/math__magicians/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>,
  <div
    style={
      Object {
        "margin": "3em",
        "padding": "2em",
      }
    }
  >
    <h2
      style={
        Object {
          "fontSize": "2rem",
        }
      }
    >
      Welcome to our page!
    </h2>
    <p
      style={
        Object {
          "margin": "3rem 0",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus elementum justo velit. Proin dui arcu, feugiat eget tellus vel, porta dapibus purus. Nam pulvinar metus sem, a ultricies ante volutpat ac. Donec ac massa ac metus lacinia molestie eget non erat. Curabitur ac turpis ut ex rutrum scelerisque. Proin tempus velit in magna aliquet fringilla. Aliquam mollis varius magna at placerat. Etiam laoreet luctus dui pharetra placerat. Sed sed neque mauris. Maecenas lorem nisi, pharetra in metus vel, tincidunt sagittis tellus. In et tempor risus. Nunc molestie tellus at sem accumsan, in vulputate tortor blandit. Sed non consectetur leo. Phasellus cursus consequat semper. Nunc feugiat consequat facilisis. Donec tempus vestibulum nisi, a blandit velit auctor nec.
    </p>
    <p
      style={
        Object {
          "margin": "3rem 0",
        }
      }
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus elementum justo velit. Proin dui arcu, feugiat eget tellus vel, porta dapibus purus. Nam pulvinar metus sem, a ultricies ante volutpat ac. Donec ac massa ac metus lacinia molestie eget non erat. Curabitur ac turpis ut ex rutrum scelerisque. Proin tempus velit in magna aliquet fringilla. Aliquam mollis varius magna at placerat. Etiam laoreet luctus dui pharetra placerat. Sed sed neque mauris. Maecenas lorem nisi, pharetra in metus vel, tincidunt sagittis tellus. In et tempor risus. Nunc molestie tellus at sem accumsan, in vulputate tortor blandit. Sed non consectetur leo. Phasellus cursus consequat semper. Nunc feugiat consequat facilisis. Donec tempus vestibulum nisi, a blandit velit auctor nec.
    </p>
  </div>,
]
`;
